Reorganise some recipes + add ffmpeg basics & filtergraph command

kfrn opened the pull request:
- Filtergraph explanation added in new section, **FFmpeg concepts**
- New section, **Create an animated GIF**, with relevant recipes moved there
- _Modify speed_ moved to **Change video properties**, since I consider speed a pretty fundamental video property.

This addresses #233, and some of the things I mentioned in #237.

kfrn committed:
Reorg items: section for GIFs, 'Modify speed' under 'Change Properties'

kfrn committed:
Add filtergraph explanation in new ffmpeg concepts section

retokromer commented:
The problem I see is that now at the beginning are the _FFmpeg concepts_ which starts with the _Filtergraph_ explanation, a pretty advanced topic. In my opinion, didactically speaking, the starting should be much gentler (e.g. on [FFmpeg’s website](https://ffmpeg.org/) they start with `ffmpeg -i input.mp4 output.avi`).

kfrn commented:
That's a good point, @retokromer. Do we call this **Advanced/Further concepts** instead and move the section to the bottom of the page? Or keep it up top, adding before the _Filtergraph_ explanation a _Basic structure of an ffmpeg command_ entry? I certainly don't want to scare away any people new to ffmpeg.

retokromer reviewed (commented):
I suggest not to start with the filtergraphs, but to move this down.

retokromer commented:
I guess _advanced concepts_ should be fine. But I would not start with this.

retokromer commented:
Adding the _basic structure_ at the beginning is excellent indeed! Yet I would still separate between basic and advanced.

kieranjol commented on index.html:
```diff
@@ -62,6 +64,41 @@
+          <p>The use of a filter is signalled by the flag <code>-vf</code> (video filter) or <code>-af</code> (audio filter), followed by the name and options of the filter itself. For example, take the <a href="#convert-colourspace">convert colourspace</a> command:</p>
+          <p><code>ffmpeg -i <i>input_file</i> -c:v libx264 -vf colormatrix=<i>src</i>:<i>dst</i> <i>output_file</i></code>
+          <p>Here, <a href="http://ffmpeg.org/ffmpeg-filters.html#colormatrix">colormatrix</a> is the filter used, with <i>src</i> and <i>dst</i> representing the source and destination colourspaces. This part following the <code>-vf</code> is a <b>filtergraph</b>.</p>
+          <p>It is also possible to apply multiple filters to an input, which are sequenced together in the filtergraph. A chained set of filters is called a filter chain, and a filtergraph may include multiple filter chains. Filters in a filterchain are separated from each other by commas (<code>,</code>), and filterchains are separated from each other by semicolons (<code>;</code>). For example ...
```
The filtergraph section looks great to me, very helpful and easy to understand. My only comment here is that the order of filters is also significant (i think) - for example, it seems to be recommended that interlaced material is deinterlaced prior to a rescale.

kfrn replied:
Oh good catch, I had meant to say something about filter order being important. Fixing this!

kfrn committed:
Add basic command explanation; relocate filtergraph explanation

- New section at top: 'Getting started with ffmpeg'. Added basic command explanation under this heading.
- New section at bottom: 'Advanced ffmpeg concepts'. Filtergraph explanation moved there.

kfrn commented:
I took your advice, @retokromer - we now have basic and advanced sections.  
I found the writing the _Basic structure_ explanation a bit tricky, terminology-wise, so I welcome any suggestions on that.

kieranjol commented on index.html:
```diff
@@ -42,26 +42,60 @@
+            <dt><i>output_file.ext</i></dt><dd>path and name of the output file.<br>
+              Because this is the last part of the command, the filename you type here does not have a flag designating it as the output file.</dd>
+          </dl>
+
```
is this space needed?

kieranjol commented on index.html:
```diff
@@ -42,26 +42,60 @@
+    <div class="modal-dialog modal-lg">
+      <div class="modal-content">
+        <div class="well">
+          <h3>Basic structure of an ffmpeg command</h3>
```
nit: for consistency - I think we were adhering to the command line argument being lowercase ffmpeg, the software itself as FFmpeg?

kfrn replied:
This is inconsistent throughout, so in my new commit I've tried to harmonise across the whole app.

kieranjol commented on index.html:
```diff
@@ -42,26 +42,60 @@
+      <div class="modal-content">
+        <div class="well">
+          <h3>Basic structure of an ffmpeg command</h3>
+          <p>At its basis, an ffmpeg command is relatively simple. After you have installed ffmpeg (see instructions <a href="https://avpres.net/FFmpeg/#ch1">here</a>), the program is invoked simply by typing <code>ffmpeg</code> at the command prompt.</p>
```
another FFmpeg - i won't make any more comments about this as i don't want to clog the feed.

kieranjol commented on index.html:
```diff
@@ -42,26 +42,60 @@
+          <dl>
+            <dt>ffmpeg</dt><dd>starts the command</dd>
+            <dt>-i <i>input_file.ext</i></dt><dd>path and name of the input file<br></dd>
+            <dt><i>-flag some_action</i></dt><dd>tell ffmpeg to do something, by supplying a valid flag and action</dd>
```
I was wondering if an example could be used here like -c:v prores, but then we'd have to explain that too, so maybe this is all just perfect as is.

kfrn replied:
I did write that exact example, but then took it out - readding!

retokromer replied:
ProRes is a little tricky, because Apple stopped supporting QuickTime on Window, and an archive should use Matroska instead. (More on this in Vienna.)

kieranjol commented:
I added a few comments, this is yet another great contribution Katherine.

kfrn commented:
I can't see them, @kieranjol - maybe you did not submit the review? I've been caught out by that before.

kieranjol commented:
Aye,sorry Katherinem Should be OK now?

On 7 Oct 2017 10:37 a.m., "Katherine Frances Nagels" <
notifications@github.com> wrote:

> I can't see them, @kieranjol <https://github.com/kieranjol> - maybe you
> did not submit the review? I've been caught out by that before.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/amiaopensource/ffmprovisr/pull/240#issuecomment-334922998>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEyvlPVYqiFrYI_oEfB8ZQ_KW2UXGNoks5sp0ZlgaJpZM4PxNhk>
> .
>

kfrn committed:
Improve descriptions; ffmpeg → FFmpeg as appropriate

Consistently use FFmpeg to refer to the program and ffmpeg for the command line invocation of the program.
Add another example to the 'asic structure of an FFmpeg command explanation.
Add a note on filter order to the Filtergraph explanation.

kfrn committed:
Add target="_blank"s to links; correct to Set Stream Properties recipe

I've replaced the links in the Set Stream Properties recipe with a more specific link.
Also, according to the docs, the correct flag is -color_primaries, not -color_primary.

kfrn commented:
Thanks heaps for the feedback @kieranjol, I've made those changes, plus a couple of other things that I caught.

I'm actually not in favour of using <i>target="_blank"</i>, but we might as well be consistent.

ablwr commented:
Wowwwwwwwwwwwww, what a treat to wake up to!

ablwr commented:
Use of "Getting started with FFmpeg" and "Learn about the basics of FFmpeg" is inconsistent -- better to use one or the other. I would also put the advanced concepts section under the Basics one, grouping them semantically by their larger theme, rather than having the advanced one get lost at the bottom.

ablwr commented:
Katherine, this is really really amazing!

retokromer commented:
I started using `target="_blank"` for external links, in order to have still the original window. If you don’t wish this behaviour, please remove them.

kfrn committed:
Both info sections at top of page; consistent naming

kfrn commented:
@ablwr - agreed and done! 

@retokromer I'm one of those people who thinks that <i>target="_blank"</i> is bad UX. But it's a minor thing, I'm not pushing for a wholesale change.

kfrn commented:
Mergin' my own pull requests, yeeeeaaaaaaa
